widgetastic_manageiq/__init__.py

```diff
@@ -1678,6 +1678,150 @@ def read(self):
+            return next(btn.title for btn in self._view_buttons if btn.active)
+        else:
+            return None
+
+    def read(self):
+        return self.selected
+
+    @property
+    def is_displayed(self):
+        return self.all_values_button.is_displayed
+
+
+class CompareToolBarModeSelector(View):
+    """ represents toolbar's Mode selector control
+        it is present on pages of Compare Selected items
+
+    .. code-block:: python
+
+        mode_selector = View.nested(CompareToolBarModeSelector)
+
+        mode_selector.select('Exists Mode')
+        mode_selector.selected
+    """
+    ROOT = './/div[contains(@class, "toolbar-pf-actions")]'
+
+    details_mode = Button(title="Details Mode")
+    exists_mode = Button(title="Exists Mode")
+
+    @property
+    def _view_buttons(self):
+        yield self.details_mode
+        yield self.exists_mode
+
+    def select(self, title):
+        for button in self._view_buttons:
+            if button.title == title:
+                return button.click()
+        else:
+            raise ValueError("The Mode with title {title} isn't present".format(title=title))
+
+    @property
+    def selected(self):
+        if self.is_displayed:
+            return next(btn.title for btn in self._view_buttons if btn.active)
+        else:
+            return None
+
+    def read(self):
+        return self.selected
+
+    @property
+    def is_displayed(self):
+        return self.details_mode.is_displayed
+
+
+class CompareToolBarViewSelector(View):
+    """ represents toolbar's view selector control
+        it is present on pages of Compare Selected items
+
+    .. code-block:: python
+
+        view_selector = View.nested(CompareToolBarViewSelector)
+
+        view_selector.select('Compressed View')
+        view_selector.selected
+    """
+    ROOT = './/div[contains(@class, "toolbar-pf-actions")]'
+
+    Expanded_button = Button(title="Expanded View")
+    Compressed_button = Button(title="Compressed View")
+
+    @property
+    def _view_buttons(self):
+        yield self.Expanded_button
+        yield self.Compressed_button
+
+    def select(self, title):
+        for button in self._view_buttons:
+            if button.title == title:
+                return button.click()
+        else:
+            raise ValueError("The view with title {title} isn't present".format(title=title))
+
+    @property
+    def selected(self):
+        if self.is_displayed:
+            return next(btn.title for btn in self._view_buttons if btn.active)
+        else:
+            return None
+
+    def read(self):
+        return self.selected
+
+    @property
+    def is_displayed(self):
+        return self.Expanded_button.is_displayed
+
+
+class CompareToolbarActions(View):
+    """ represents compare toolbar's all actions
+        it is present on pages of Compare Selected items and drift
+
+    .. code-block:: python
+
+        actions = View.nested(CompareToolbarActions)
```

Comment:
whether all these nested views can exist w/o this Toolbar ?
I'd merge all these things in one view if not.

Reply:
yes @izapolsk , What I found up to now. All three selector always share the same toolbar ["compare selected items" and "drift"]. I thought if anyone wants to use separately, separation is good rather than add in a single class.

Reply:
I don't think there will be places where those are used separately.
I think you need to merge all of them into one main view with several nested views under it.
you will get rid of most duplicate code then.